### DISCUSSION TITLE
Alternative AST Querying Languages: `rese/astq`

### DISCUSSION THREAD

**Discussion by chriscz** [ast-grep/ast-grep#1445] [Ideas]
Hi, I just began playing with and using `ast-grep`. I love the power of this tool and funnily  enough I've been telling my team how badly I want something like this.

In a recent web search I came across the following querying language and thought it might be interesting / useful to consider it in future design: https://github.com/rse/astq

**Comment by HerringtonDarkholme** [comment]
Hi Chris,

Thank you for your suggestion.

ast-grep has been primarily inspired by CSS syntax from its inception. The astq, however, draws its inspiration from XPath. While it is possible that future versions of ast-grep may introduce a CSS-selector-like path, this is not a high priority at the moment.
